### ISSUE TITLE
yarn global doesn't always create bin symlinks

### ISSUE THREAD

**Issue by erbridge** [yarnpkg/yarn#2360]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Some package binaries aren't symlinked into `yarn global bin` as they should be. They are in `~/.config/yarn/global/node_modules/.bin`, however.

**If the current behavior is a bug, please provide the steps to reproduce.**

```sh
$ ls $(yarn global bin)
# does not contain torus or torus-cli

$ yarn global add torus-cli

$ ls $(yarn global bin)
# still does not contain torus or torus-cli

$ ls ~/.config/yarn/global/node_modules/.bin
# does contain torus and torus-cli
```

**What is the expected behavior?**

`yarn global add` should create symlinks in `yarn global bin`.

**Please mention your node.js, yarn and operating system version.**

- node v7.3.0
- yarn v0.18.1
- Arch Linux

**Comment by chnn** [comment]
I am having the same issue, with the same version of node and yarn on macOS. Installing the `ember-cli` package does not create an `ember` symlink in the global bin path. Oddly enough though, installing `bower` does create a `bower` symlink in the global bin path.

**Comment by JaKXz** [comment]
Also a duplicate of #1183 by the looks of it.

**Comment by pbasista** [comment]
No, #1183 does not seem to be related, because it deals with the actual execution of the installed package, not with its linking to yarn's global bin directory.

**Comment by pbasista** [comment]
Another package which fails to install properly (i.e. with symlinks in the global `bin` directory) is `eslint`.

**Comment by pbasista** [comment]
It is also *not* a duplicate of #1585, because that one deals with the behavior of `yarn link`, while this issue deals with the behavior of `yarn global add`.

**Comment by pbasista** [comment]
I am unable to reproduce this issue on linux with `torus-cli` package:
```
$  yarn global add torus-cli
yarn global v0.27.5
warning package.json: No license field
warning No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Installed "torus-cli@0.24.1" with binaries:
      - torus
      - torus-cli
warning No license field
Done in 3.52s.
```
The scripts are in place:
```
$ ls $(yarn global bin)
torus  torus-cli
```

**Comment by pbasista** [comment]
I was able to reproduce this with `eslint` a while ago, but not anymore. I experimented with adding, removing, linking and unlinking of various packages in the meantime. Now it works for me with both `torus-cli` and `eslint` packages, so I am confused.

The described behavior is definitely there, but I am not sure how to reliably trigger it.

**Comment by milenvoutchev** [comment]
I just reproduced this issue as well. When installing `serve`:
`yarn global add serve` reports successfully installed:
```
success Installed "serve@6.0.3" with binaries:
      - serve
```
However, the symlink has actually not been created:
```
$ serve
-bash: /usr/local/bin/serve: No such file or directory
```
I'm running:
```yarn version v0.27.5
macOS Sierra 10.12.5```

**Comment by alexreg** [comment]
@rally25rs @arcanis Any update on this, kindly? It seems like a fairly high-priority issue that has been lingering for some time now... there's also https://github.com/yarnpkg/yarn/issues/4964, which seems closely related.
